docs/source/articles/decorate-and-write-tests.html.md

```diff
@@ -129,6 +129,70 @@ module Workarea
 end
 ```
 
+### Change Setup/Teardown Behavior
+
+To change how tests are set up or torn down between each individual run, use the `setup` and `teardown` DSL methods:
+
+```ruby
+module Workarea
+  class ImportInventoryTest < TestCase
+    setup :setup_custom_logic
+    teardown :teardown_custom_logic
+
+    def test_perform
+      # ...
+    end
+
+    def setup_custom_user_logic
+      # do your custom setup logic here
+    end
+
+    def teardown_custom_user_logic
+      # do your custom teardown logic here
+    end
+  end
+end
+```
+
+Applications may also use the block style for these DSL methods, but **plugins should not use this style** as it is impossible to decorate in an application:
+
+```ruby
+module Workarea
+  class ImportInventoryTest < TestCase
+    setup do
+      # do your custom setup logic here
+    end
+
+    teardown do
+      # do your custom teardown logic here
+    end
+
+    def test_perform
+      # ...
+    end
+  end
+end
+```
+
+In Workarea 3.5.0, `Workarea::TestCase` will do a few cleanup tasks for you on teardown, so writing tests like this won't accidentally leak configuration and cause random test failures:
```

Comment:
```suggestion
As of Workarea 3.5.0, `Workarea::TestCase` will do a few cleanup tasks for you on teardown, so writing tests like this won't accidentally leak configuration and cause random test failures:
```

This will be true going forward.